Sources/IdentityState.swift

```diff
@@ -141,17 +167,22 @@ class IdentityState {
 
     /// Clears all identities and regenerates a new ECID value.
     /// Saves identities to persistence and creates a new XDM shared state and dispatches a new` resetComplete` event after operation completes.
+    /// Dispatches Consent request event setting `adId` to 'no' if previous advertising identifier is nil or empty
     /// - Parameters:
     ///   - event: event which triggered the reset call
     ///   - createXDMSharedState: function which creates new XDM shared states
     ///   - eventDispatcher: function which dispatches a new `Event`
     func resetIdentifiers(event: Event,
                           createXDMSharedState: ([String: Any], Event) -> Void,
                           eventDispatcher: (Event) -> Void) {
-
+        let shouldDispatchConsent = identityProperties.advertisingIdentifier != nil && !(identityProperties.advertisingIdentifier?.isEmpty ?? true)
         identityProperties.clear()
         identityProperties.ecid = ECID().ecidString
 
+        if shouldDispatchConsent {
+            dispatchAdIdConsentRequestEvent(val: IdentityConstants.XDMKeys.Consent.NO, eventDispatcher: eventDispatcher)
```

Comment:
That makes sense; I've moved the `shouldDispatchConsent` block to before the `identityProperties` are cleared

Reply:
That won't actually solve the issue. When the event is dispatched, a new event is created and when the Edge extension handles it, it will use the event number to look up the Identities (and ECID) from the Identity shared state. During reset identities, the Identity extension will use the reset identities event to create a shared state with the new ECID. As the reset identities event is before the consent event, Edge will always get the new, reset, ECID, no mater the order of the calls here. 
We'll need to determine if sending a consent No is required before resetting an ECID. If so, we may need to come up with another solution.
CC: @emdobrin

Reply:
We can check with the upstream services and revisit as part of a new PR.